test-loop-tests/src/tests/congestion_control.rs

```diff
@@ -59,7 +65,55 @@ fn slow_test_congestion_control_simple() {
         .shutdown_and_drain_remaining_events(Duration::seconds(20));
 }
 
-fn setup(accounts: &Vec<AccountId>) -> (TestLoopEnv, AccountId) {
+#[cfg_attr(not(feature = "test_features"), ignore)]
+#[test]
+fn slow_test_one_shard_congested() {
+    init_test_logger();
+
+    let mut accounts = make_accounts(NUM_PRODUCERS + NUM_VALIDATORS);
+    let shard1_acc1: AccountId = "0000".parse().unwrap();
+    let shard1_acc2: AccountId = "1111".parse().unwrap();
+    let shard2_acc1: AccountId = "xxxx".parse().unwrap();
+    accounts.push(shard1_acc1.clone());
+    accounts.push(shard1_acc2.clone());
+    accounts.push(shard2_acc1.clone());
+
+    let (env, rpc_id) = setup(&accounts, &["near"]);
+    assert_eq!(rpc_id, shard1_acc1);
+
+    // Configure test env to drop chunks for shard 2
+    let shard2 = env.shared_state.genesis.config.shard_layout.account_id_to_shard_id(&shard2_acc1);
+    let dropped_chunks = [
+        (shard2, (0..1000).map(|_| false).collect_vec()), // Drop 1000 chunks for shard 2
+    ]
+    .into_iter()
+    .collect();
+    let mut env = env.drop(DropCondition::ChunksProducedByHeight(dropped_chunks));
+
+    // Run for two epochs to make sure shard 2 is congested
+    let client_handle = env.node_datas[0].client_sender.actor_handle();
+    let client_actor = env.test_loop.data.get(&client_handle);
+    let block_time = client_actor.client.config.max_block_production_delay;
+    env.test_loop.run_for(block_time * EPOCH_LENGTH as u32 * 2);
```

Comment:
This is a bit risky, I'd be better to assert directly on the number of blocks rather than the time.

Reply:
Can you assert that shard 1 is indeed congested?

Reply:
> Can you assert that shard 1 is indeed congested?

No idea how to do it.

Reply:
block should have congestion info in it, the chunks headers probably also have it